lib/porro/types/default.rb

```diff
@@ -13,11 +13,11 @@ def initialize(wrapper, default = nil)
       end
 
       def load(value)
-        @wrapper.load(value) || default_proc.call
+        @wrapper.load(value) || @wrapper.load(default_proc.call)
       end
 
       def dump(attribute)
-        @wrapper.dump(attribute) || default_proc.call
+        @wrapper.dump(attribute) || @wrapper.dump(default_proc.call)
```

Comment:
This is the counter-part, especially for cases like

`attribute :birthday, Porro.date.default(-> { Time.now })`

The short-circuit from previously will behave like:

`Model.new.dump => { ..., birthday: #<Date: 2016-04-08 ((2456041j,0s,0n),+0s,2299161j)> }` 

but what we would expect is something like ` { ..., birthday: "2016-04-08T00:00:00.000Z" }`.

The alternative `@wrapper.dump(attribute || default_proc.call)` is prone to the same issue as described above in the `load` case.